server/src/main/java/org/elasticsearch/index/engine/TranslogDirectoryReader.java

```diff
@@ -417,21 +432,7 @@ public int maxDoc() {
 
         @Override
         public void document(int docID, StoredFieldVisitor visitor) throws IOException {
-            assert docID == 0;
-            if (docID != 0) {
-                throw new IllegalArgumentException("no such doc ID " + docID);
-            }
```

Comment:
I noticed this check is now gone. Is this on purpose?

Reply:
Given that this is an internal class, it seems like overkill to have both the assertion and the exception?  @dnhatn what do you think?

Reply:
OK, I just didn't know all the places where this could be used. I just saw that this was no longer a simple find/replace. 

I see how this is used in memory "real-time" look up. DocID there doesn't really make sense at all. So, it should be ok.

Reply:
Either way works for me. We already have this check when creating the reader: 
https://github.com/elastic/elasticsearch-internal/blob/c9b2810da4a9db62e5f43a94f3cf49d9772fb036/server/src/main/java/org/elasticsearch/index/engine/TranslogDirectoryReader.java#L270